In [1]:
import pandas as pd

df = pd.read_csv('data/human_body_temperature.csv')

In [2]:
# Your work here.

## Preliminary look at the data

In [3]:
df.head()

temperature gender  heart_rate
0         99.3      F        68.0
1         98.4      F        81.0
2         97.8      M        73.0
3         99.2      F        66.0
4         98.0      F        73.0

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 130 entries, 0 to 129
Data columns (total 3 columns):
temperature    130 non-null float64
gender         130 non-null object
heart_rate     130 non-null float64
dtypes: float64(2), object(1)
memory usage: 3.1+ KB


In [5]:
df['temperature'].describe()

count    130.000000
mean      98.249231
std        0.733183
min       96.300000
25%       97.800000
50%       98.300000
75%       98.700000
max      100.800000
Name: temperature, dtype: float64

In [6]:
df['gender'].value_counts()

F    65
M    65
Name: gender, dtype: int64

## 1) Is the distribution of body temperatures normal?

In [7]:
#run scipy's built in test for normal distribution
from scipy import stats

In [8]:
scipy.stats.normaltest(df['temperature'])

NameError: name 'scipy' is not defined

With the Null Hypothesis that the sample distribution is drawn from a normally distributed population, the p-vaule result above of 0.25 is well above the traditional threshold of 0.05 that would be used to reject the Null, and so conclude that the distribution of body temperatures is likely to be normal.

## 2) Is the sample size large? Are the observations independent?

The sample size of n = 130 is well above the benchmark of n = 30, which is considered to be sufficiently large. The dataset description gives no indication of any relationship among the individuals sampled and so the observations are considered independent also.

## 3) Is the true population mean really 98.6 degrees F?

For this I would use a one-sample test as we are just testing a statistic from one sample against the associated population statistic, rather than test statistics from two samples against one another.

Although the population variance is unknown, for the test it is appropriate to use the *z* statistic as the sample size is greater than 30. With a sample of this size the *z* and *t* distributions look pretty much the same and give roughly the same result.

Null Hypothesis: pop mean = 98.6
Alternative Hypothesis: pop mean != 98.6

z = (98.25 - 98.6) / (0.73 / Sqrt(130))
  = - 5.47
  
The mean of our sample is approximately 5.5 standard deviations below the mean of the sampling distribution. This number of standard deviations is too high for the z table meaning that even with a confidence level of over 99.99% we can reject the Null Hypothesis of the pop mean being 98.6.

The t-test would use the same calculation and similarly the t score does not appear on the conventional table even at a 99.9% confidence level.

In conclusion, we do not know what the true population mean really is. However, if it is 98.6 then there is an extremely low probability that we should get a sample with a mean as low as the one above.

## 4) At what temperature should we consider someone's temperature to be "abnormal"? 
For this we will calculate a 95% confidence interval. Values falling outside will be considered "abnormal". 

Confidence level of 95% equates to a critical level of 1.96.

1.96 * (0.73 / Sqrt(130)) = 0.125 = Margin of Error

95% Confidence Interval using the sample mean (98.125, 98.375)
Temperatures falling outside these bounds may be considered "abnormal".

## 5) Is there a significant difference between males and females in normal temperature?

For this I would use the following two-sample test:

Null Hypothesis: male pop mean - female pop mean = 0

Alternative Hypothesis: male pop mean - female pop mean != 0

The sample sizes for both males and females is 65 and so large enough to use *z* test scores.

I would test for a significance level of 5%.

In [ ]:
#calculate sample statistics for males
df[df['gender'] == 'M']['temperature'].describe()

In [ ]:
#calculate sample statistics for females
df[df['gender'] == 'F']['temperature'].describe()

Difference in sample means is 0.289.

Difference in means for sampling distribution is 0 under null hypothesis.

Standard Error = Sqrt((0.699^2 / 65) + (0.743^2 / 65))
               = 0.1265
               
Critical value = 0.1265*1.96 = 0.248

The difference in the sample means exceeds the critical value at the 5% significance level meaning that we reject the Null Hypothesis that there is no difference between average male and female body temperature under our chosen significance level.

Given the original question and our conclusion that females have a higher average body temperature, perhaps it is possible that the original estimation of a population mena of 98.6 contained a dispropotionate amount of females in the sampling. Or perhaps more likely that the measurements used in the sampling were inexact, particularly given the age of the original report.